### ISSUE TITLE
Can't create uberjar with leiningen

### ISSUE THREAD

**Issue by rickerbh** [nilenso/goose#96]
I'm hoping this is an issue with the way I'm attempting to use this, but I've found an issue with creating an uberjar via leiningen that include a goose worker. The issue seems to be that including the `goose.worker` namespace in a project that creates an uberjar, causes the metrics plugin to be initialised (as it's in a `def`), and the LOC for the metrics plugin at https://github.com/nilenso/goose/blob/23a60110f37e16fddcb7b0ceeeb7bc87d9766f2e/src/goose/metrics/statsd.clj#L67 gets called and this causes compilation to halt. 

You can work around it by replicating the worker namespace in your own project, and passing `{:enabled? false}` like below
`(statsd/new (merge statsd/default-opts {:enabled? false}))`. However, you can't avoid the initialisation of `goose.worker/default-opts`, so any inclusion of the worker namespace causes this issue. This isn't a problem from the repl, only with uberjar creation.

Is there a way around this? Am I holding it wrong? 😉 

I've embedded the files for a minimal reproducing case below.

Create project with `lein new app gtest`, then replace below files

`project.clj`
```
(defproject gtest "0.1.0-SNAPSHOT"
  :description "FIXME: write description"
  :url "http://example.com/FIXME"
  :license {:name "EPL-2.0 OR GPL-2.0-or-later WITH Classpath-exception-2.0"
            :url "https://www.eclipse.org/legal/epl-2.0/"}
  :dependencies [[org.clojure/clojure "1.11.1"]
                 [com.nilenso/goose "0.3.0"]]
  :main ^:skip-aot gtest.core
  :target-path "target/%s"
  :profiles {:uberjar {:aot :all
                       :jvm-opts ["-Dclojure.compiler.direct-linking=true"]}})
```

`core.clj`
```
(ns gtest.core
  (:require [goose.worker :as w])
  (:gen-class))

(defn -main
  "I don't do a whole lot ... yet."
  [& args]
  (println "Hello, World!"))
```

`lein uberjar` - starts to compile then halts.

**Comment by rickerbh** [comment]
Investigating more, the `*compile-files*` flag might help here, to stop evaluation of the `def`. I'm happy to submit a PR if that's an appropriate solution.

**Comment by olttwa** [comment]
An additional thing to note here should be that uberjar creation works with Clojure tools.build.

I referred to [Volodymyr Kozieiev's blog on creating uberjars using deps.edn] & was able to compile & run my program.

Please refer to [this Github Gist](https://gist.github.com/olttwa/9eb9bdb14f55fc66e2e7010057fc4cbd) for reference code to build & run an uberjar

**Comment by rickerbh** [comment]
@olttwa Yes, the removal of the metrics plugin init from the def sounds like a good solution to me. 

Thanks for the link to that article too. I do want to reduce our dependencies on leiningen, so this is super helpful.

**Comment by olttwa** [comment]
@rickerbh We've released `0.3.1` with a fix.
I tried creating an uberjar with lein and it's working now 🥳 

Please go ahead with your spike! If you face any hiccups, or have concerns using it in Production, feel free to create another issue, or ping us on [#goose @Clojurians slack](https://clojurians.slack.com/channels/goose).
